samples/compression/lz4/prj.conf

```diff
@@ -0,0 +1,4 @@
+CONFIG_LZ4=y
+CONFIG_NEWLIB_LIBC=y
+
+CONFIG_HEAP_MEM_POOL_SIZE=16384
```

Comment:
There are 2 possibilities:
- Just leave `CONFIG_NEWLIB_LIBC=y` and remove `CONFIG_HEAP_MEM_POOL_SIZE=16384` (won't be needed after dropping `k_malloc()`).
- Use minimal libc (by removing `CONFIG_NEWLIB_LIBC=y`) and add `CONFIG_MINIMAL_LIBC_MALLOC_ARENA_SIZE=24576` (24kB, where ~17kB goes for `LZ4_stream_t` internal LZ4 allocation on heap and 2-3kB goes for application data that is compressed/uncompressed).

I'd prefer minimal libc, as it is smaller in size.